### DISCUSSION TITLE
Is it possible to test the order of items in a list?

### DISCUSSION THREAD

**Discussion by lukebrandonfarrell** [wix/Detox#2984] [Q&A]
Question is in the title. How would you test order of items in a list using Detox? For our scenario we have a list which can be re-ordered using a selection picker UI. We want to test that the re-order is working.

Any ideas?

**Comment by lukebrandonfarrell** [comment]
I have a possible implementation for this, where you can use a testID with a dynamic index to test order of items. Will share in the coming weeks.

**Reply by d4vidi** [reply]
That was where I was going for with an possible suggestion. But indeed, you'd have to apply identical test ID's (or otherwise have Detox support ID's wildcards/regex's).

**Reply by d4vidi** [reply]
Alternatively, screenshot testing can be applied here as well. We even have an API to take screenshots of specific screen elements (and their child-views - meaning, it works for lists as well).

https://github.com/wix/Detox/blob/master/docs/APIRef.Screenshots.md
